docs/user_guide/6_migration_guide/index.rst

```diff
@@ -0,0 +1,108 @@
+- `global`: Defines model-wide configuration, including components and data libraries.
+- `steps`: Replaces the old numbered dictionary format with a sequential list of function calls.
+
+Functions are now explicitly mapped to model or component methods using the `<component>.<method>` syntax.
+
+For a complete example of the new configuration format, see the Wflow v1 YAML template: :download:`wflow_build.yml <../../_examples/wflow_build.yml>`.
+
+For a more information on the format changes, see this section in the HydroMT migration guide: `Changes to the yaml HydroMT configuration file format <https://deltares.github.io/hydromt/stable/guides/plugin_dev/migrating_to_v1.html#changes-to-the-yaml-hydromt-configuration-file-format>`_.
+
+
+Data Catalog Format Changes
+---------------------------
+
+The data catalog structure has been refactored to introduce a more modular design and clearer separation of responsibilities across several new classes (`DataSource`, `Driver`, `URIResolver`, and `DataAdapter`).
+
+Key format changes:
+
+- `path` → renamed to `uri`
+- `filesystem` moved under `driver`
+- `unit_add`, `unit_mult`, `rename`, etc. moved under `data_adapter`
+- A single catalog entry can now reference multiple data variants or versions
+
+For detailed information about the format changes, see this section in the HydroMT migration guide: `Changes to the data catalog yaml file format <https://deltares.github.io/hydromt/stable/guides/plugin_dev/migrating_to_v1.html#datacatalog>`_
+
+Removing Dictionary-like Catalog Access
+---------------------------------------
+
+Catalog access has changed from dictionary-style lookups to an explicit API.
+Update your code accordingly:
+
++------------------------------+------------------------------------------+
+| v0.x                         | v1                                       |
++==============================+==========================================+
+| `catalog["name"]`            | `catalog.get_source("name")`             |
++------------------------------+------------------------------------------+
+| `'name' in catalog`          | `catalog.contains_source("name")`        |
++------------------------------+------------------------------------------+
+| `catalog.keys()`             | `catalog.get_source_names()`             |
++------------------------------+------------------------------------------+
+| `catalog["name"] = data`     | `catalog.set_source("name", data)`       |
++------------------------------+------------------------------------------+
+
+
+Migration to Wflow v1
+=====================
+
+The Wflow v1 update introduces new model configuration parameters, renamed parameters and upgraded file structures consistent with HydroMT Core v1.
+
+To convert an existing v0.x hydromt wflow sbm model, you can use the cli command:
+
+.. code-block:: bash
+
+   hydromt update <model_type> <model_root> -i <upgrade_v1.yml> -v
```

Comment:
Hum... I see in the example I went the way with staticmaps and config getting a new name. Maybe here we could go with the -o option <model_root_v1>
The notebooks by the way did not run in the html zip you sent